libp2p/NodeTable.cpp

```diff
@@ -233,25 +273,9 @@ void NodeTable::doDiscover(NodeID _node, unsigned _round, shared_ptr<set<shared_
         return;
     }
 
-    m_timers.schedule(c_reqTimeout.count() * 2, [this, _node, _round, _tried](boost::system::error_code const& _ec)
-    {
-        if (_ec)
-            // we can't use m_logger here, because captured this might be already destroyed
-            clog(VerbosityDebug, "discov")
-                << "Discovery timer was probably cancelled: " << _ec.value() << " "
-                << _ec.message();
-
-        if (_ec.value() == boost::asio::error::operation_aborted || m_timers.isStopped())
-            return;
-
-        // error::operation_aborted means that the timer was probably aborted.
-        // It usually happens when "this" object is deallocated, in which case
-        // subsequent call to doDiscover() would cause a crash. We can not rely on
-        // m_timers.isStopped(), because "this" pointer was captured by the lambda,
-        // and therefore, in case of deallocation m_timers object no longer exists.
-
-        doDiscover(_node, _round + 1, _tried);
-    });
+    _timer->expires_from_now(boost::posix_time::milliseconds(c_reqTimeout.count() * 2));
```

Comment:
Maybe make a constant out of `boost::posix_time::milliseconds(c_reqTimeout.count() * 2)`